04/app.js

```diff
@@ -1,89 +1,104 @@
-import React from 'react';
-import { createRoot } from 'react-dom/client';
+import React from "react";
+import { createRoot } from "react-dom/client";
 
-const root = createRoot(document.querySelector('#root'));
+const root = createRoot(document.querySelector("#root"));
 
 class App extends React.Component {
-    state = { 
-        firstName: '',
-        lastName: '',
-        searchQuery: '',
-        users: ['Jan Kowalski', 'Michał Nowak'],
-    }
+  state = {
+    firstName: "",
+    lastName: "",
+    searchQuery: "",
+    users: ["Jan Kowalski", "Michał Nowak"],
+  };
 
-    renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
-                <li onClick={ this.clickHandler }>
-                    { name }
-                </li>
-            );
-        });
-    }
+  renderUsersList() {
+    const { searchQuery, users } = this.state;
 
-    clickHandler = e => {
-        const {innerText: userName} = e.target;
-        this.removeUser(userName);
+    if (searchQuery) {
+      return users.map((name) => {
```

Comment:
Tutaj powinnaś ustawić filter.
W zmiennej `users` masz wszystkich użytowników i przez `.map()` ich renderujesz.
Ty chcesz ograniczyć liczbę użytkoników więc musisz to zrobić przed `map()` czyli:
`return users.filter(item => item.includes(searchQuery)).map(`

PS. Warto też tak zrobić, aby wielkość liter nie miała znaczenia - wystarczy zamienić "oba teksty" na małą literę.